### PR TITLE
Create org.pitivi.Pitivi.json

### PR DESCRIPTION
**Pull request by thiblahute**

### CODE REVIEW

File: org.pitivi.Pitivi.json [hunk 1]
```diff
@@ -0,0 +1,294 @@
+{
+    "app-id": "org.pitivi.Pitivi",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.22",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: Any reason why not 3.24?

File: org.pitivi.Pitivi.json [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+    "finish-args": ["--command=pitivi",
+                    "--share=ipc",
+                    "--socket=x11",
+                    "--socket=session-bus",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: Ideally only grant what you need rather than blanket access.

File: org.pitivi.Pitivi.json [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+        "cflags": "-O0 -g",
+        "cxxflags": "-O0 -g",
+        "env": {
+          "PYTHON": "python3",
+          "GST_PLUGIN_SYSTEM_PATH": "/app/lib/gstreamer-1.0/"
+        }
+    },
+    "modules": [
+        {
+            "name": "yasm",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: This is in the Sdk.

File: org.pitivi.Pitivi.json [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+                    "sha256": "5d98e9e4faf6dd55e7193ed379aff477b8acbda6777758956ef7e5f05067be18"
+                }
+            ]
+        },
+        {
+            "name": "numpy",
+            "build-options": {
+              "build-args": ["--share=network"],
+              "cflags": "-O0 -g  -L/usr/lib -Lbuild/temp.linux-x86_64-3.4 -I/usr/include -I/usr/include/python3.4m/",
+              "env": {"ARCHFLAGS": "-arch x86_64"}
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: Flathub supports multiple arches so hardcoding this is wrong unless you only support that one arch.

File: org.pitivi.Pitivi.json [hunk 5]
```diff
@@ -0,0 +1,294 @@
...
+                    "type": "archive",
+                    "url": "http://download.videolan.org/pub/x264/snapshots/x264-snapshot-20140212-2245-stable.tar.bz2",
+                    "sha256": "5d98e9e4faf6dd55e7193ed379aff477b8acbda6777758956ef7e5f05067be18"
+                }
+            ]
+        },
+        {
+            "name": "numpy",
+            "build-options": {
+              "build-args": ["--share=network"],
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: Flathub is sandboxed, you cannot grant network access. You have to fix their build system to work offline.

File: org.pitivi.Pitivi.json [hunk 6]
```diff
@@ -0,0 +1,294 @@
...
+            "sources": [
+                {
+                    "type": "file",
+                    "path": "py-configure",
+                    "dest-filename": "configure"
+                }
+            ]
+        },
+        {
+            "name": "meson",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: This is in the Sdk.

File: org.pitivi.Pitivi.json [hunk 7]
```diff
@@ -0,0 +1,294 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://bitbucket.org/multicoreware/x265/downloads/x265_1.9.tar.gz",
+                    "sha256": "3e4654133ed957a98708fdb4cb9a154d9e80922b84e26e43fc462a101c5b15c8"
+                }
+            ]
+        },
+        {
+            "name": "gstreamer",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: In the 3.24 Sdk.

File: org.pitivi.Pitivi.json [hunk 8]
```diff
@@ -0,0 +1,294 @@
...
+                {
+                    "type": "git",
+                    "branch": "1.10.5",
+                    "commit": "88ab82933382442a3a626da400f0fd532229b0f9",
+                    "url": "git://anongit.freedesktop.org/gstreamer/gstreamer"
+                }
+            ]
+        },
+        {
+            "name": "gst-plugins-base",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: In the 3.24 Sdk.

File: org.pitivi.Pitivi.json [hunk 9]
```diff
@@ -0,0 +1,294 @@
...
+                {
+                    "type": "git",
+                    "branch": "1.10.5",
+                    "commit": "d6f30c28d8cfdad1199f7d29ec248a12bc26606d",
+                    "url": "git://anongit.freedesktop.org/gstreamer/gst-plugins-base"
+                }
+            ]
+        },
+        {
+            "name": "gst-plugins-good",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: In the 3.24 Sdk.

If you only need to build a few extra plugins specify so in the config-opts.

File: org.pitivi.Pitivi.json [hunk 10]
```diff
@@ -0,0 +1,294 @@
...
+                    "commit": "5c5acca68b786bdd0ccefabed8fd0209321e73af",
+                    "url": "git://anongit.freedesktop.org/gstreamer/gst-editing-services"
+                }
+            ]
+        },
+        {
+            "name": "pitivi",
+            "buildsystem": "meson",
+            "builddir": true,
+            "config-opts": ["--libdir=lib", "--prefix=/app"],
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: Don't need to pass prefix manually.

File: org.pitivi.Pitivi.json (hunk 1 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: Yes, this version has been developed and mostly tested on 3.22, only in current master we support 3.24 (fixed minor warning and stuff).

File: org.pitivi.Pitivi.json (hunk 4 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: Good point!

File: org.pitivi.Pitivi.json (hunk 5 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: Why is that? What does it change to access the network from the host or from withing the sandbox?

File: org.pitivi.Pitivi.json (hunk 6 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: Right, but we do not use 3.24 :-)

File: org.pitivi.Pitivi.json (hunk 9 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: Same.

File: org.pitivi.Pitivi.json (hunk 2 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: ipc, is the onlyone that might not be required, am I missing something else?

File: org.pitivi.Pitivi.json (hunk 4 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: Fixed

File: org.pitivi.Pitivi.json (hunk 2 above)
Comment (TingPing) [inline org.pitivi.Pitivi.json]: `--socket=session-bus` gives you unrestricted access to the dbus session. You should be using `--talk-name=foo.bar` instead.

File: org.pitivi.Pitivi.json (hunk 1 above)
Comment (TingPing) [inline org.pitivi.Pitivi.json]: I would strongly recommend 3.24 as it has had numerous improvements like supporting theming, VAAPI support, GStreamer 1.10 and other dependencies you use, etc.

File: org.pitivi.Pitivi.json (hunk 5 above)
Comment (TingPing) [inline org.pitivi.Pitivi.json]: This was mentioned on IRC but we want everything to be reproducable and it stops any possibility of any malicious network use.

File: org.pitivi.Pitivi.json (hunk 1 above)
Comment (thiblahute) [inline org.pitivi.Pitivi.json]: We do not want vaapi (not robust enough for us), our official development environment is flatpak and for a specific release we target a specific runtime/sdk version, changing randomly afterward is like what used to happen with distro, and this is really something I do not want.

**Comment by thiblahute** [comment]
Updated with the following changes:
* Nework grant `network` access to the module builds directly using setup.py
* Remove wrong ` "--share=ipc", "--socket=session-bus"` access granding
* Remove `--prefix` from pitivi target
* Add `gst-transcoder` as a module as using it as a subproject lead to a wrong libdir path installation (ie. it was installing to `lib64/`
* Remove archs definition as those were not needed (hopefully Pitivi will build on more arch :-))

**Comment by thiblahute** [comment]
Anything missing?

**Comment by Barthalion** [comment]
No. Looks good to me (although I dislike that you are using 3.22, but I see your point); I scheduled a [test build](https://flathub.org/builds/#/buildrequests/2601?redirect_to_build=true).

**Comment by TingPing** [comment]
Build failed since 3.22 isn't available on the builders.

**Comment by thiblahute** [comment]
> Build failed since 3.22 isn't available on the builders.

I tested with `3.24` and we have regression (that are fixed in master), I will try to get pitivi 0.99 released soon and update that PR.

**Comment by purpleidea** [comment]
> Build failed since 3.22 isn't available on the builders.

Why not?

**Comment by thiblahute** [comment]
> Build failed since 3.22 isn't available on the builders.
> Why not?

Also a very good question! :-)

**Comment by TingPing** [comment]
Probably because nobody needed it yet and 3.24 has a lot of improvements. I imagine Alex isn't against adding it though.

**Comment by purpleidea** [comment]
@thiblahute 
> Also a very good question! :-)

Wasn't meant as sarcasm, but rather, I thought the whole point is that flatpak would allow application authors to target different runtimes, thus not breaking apps if gnome moves faster than they do for example. Personally, I think you should just get on the latest, but I understand both sides :)

**Comment by thiblahute** [comment]
> @thiblahute
> 
> Also a very good question! :-)
> Wasn't meant as sarcasm, but rather, I thought the whole point is that flatpak would allow application authors to target different runtimes, thus not breaking apps if gnome moves faster than they do for example.

I was not taking it as sarcasm, just stating that it would also be good to have.

>Personally, I think you should just get on the latest, but I understand both sides :)

Well we have but it is not released yet.

@alexlarsson 

> Probably because nobody needed it yet and 3.24 has a lot of improvements. I imagine Alex isn't against adding it though.

(How) could we get that done?

**Comment by TingPing** [comment]
@thiblahute I'm curious where the regression you have is? Because they should have the same Gtk version (3.24 doesn't exist), they should have the same GStreamer since you build it yourself, etc.

**Comment by thiblahute** [comment]
> @thiblahute I'm curious where the regression you have is? Because they should have the same Gtk version (3.24 doesn't exist), they should have the same GStreamer since you build it yourself, etc.

We have an issue in latest meson (we could workaround that), and in Gtk between 3.22.6 (Sdk 3.22.0) and 3.22.14 in the new Sdk.

**Comment by TingPing** [comment]
> and in Gtk between 3.22.6 (Sdk 3.22.0) and 3.22.14 in the new Sdk.

Hmm, well it is probably just chance that the old Sdk didn't get those Gtk3 updates anyway. As you mention Meson is trivial to workaround.

**Comment by thiblahute** [comment]
OK, `0.99` is out based on the 3.24 runtime, please review again :-)

File: org.pitivi.Pitivi.json [hunk 11]
```diff
@@ -0,0 +1,497 @@
+{
+    "app-id": "org.pitivi.Pitivi",
+    "branch": "1.0",
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: Set it to `stable` or remove altogether, our CI sets it on its own.

File: org.pitivi.Pitivi.json [hunk 12]
```diff
@@ -0,0 +1,497 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.24",
+    "finish-args": ["--command=pitivi",
+                    "--socket=x11",
+                    "--socket=pulseaudio",
+                    "--socket=wayland",
+                    "--socket=session-bus",
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: Does it need access to entire session bus?

File: org.pitivi.Pitivi.json [hunk 13]
```diff
@@ -0,0 +1,497 @@
+{
+    "app-id": "org.pitivi.Pitivi",
+    "branch": "1.0",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.24",
+    "finish-args": ["--command=pitivi",
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: Use `command` top level property instead.

File: org.pitivi.Pitivi.json [hunk 14]
```diff
@@ -0,0 +1,497 @@
...
+                    "--talk-name=ca.desrt.dconf",
+                    "--talk-name=org.freedesktop.Notifications",
+                    "--filesystem=host", "--device=dri"],
+    "sdk": "org.gnome.Sdk",
+    "rename-appdata-file": "pitivi.appdata.xml",
+    "rename-desktop-file": "pitivi.desktop",
+    "rename-icon": "pitivi",
+    "copy-icon": true,
+    "build-options": {
+        "cflags": "-O0 -g",
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: So, a nitpick… Why not `-O2`?

File: org.pitivi.Pitivi.json [hunk 15]
```diff
@@ -0,0 +1,497 @@
...
+        "cxxflags": "-O0 -g",
+        "env": {
+          "PYTHON": "python3",
+          "GST_PLUGIN_SYSTEM_PATH": "/app/lib/gstreamer-1.0/",
+          "FREI0R_PATH": "/app/lib/frei0r-1/"
+        },
+        "strip": false,
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: It's the default.

File: org.pitivi.Pitivi.json [hunk 16]
```diff
@@ -0,0 +1,497 @@
...
+        "no-debuginfo": true
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: Ditto.

File: org.pitivi.Pitivi.json [hunk 17]
```diff
@@ -0,0 +1,497 @@
...
+    },
+    "modules": [
+        {
+            "name": "numpy",
+            "buildsystem": "simple",
+            "ensure-writable": ["easy-install.pth"],
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: @TingPing does it look right to you?

File: org.pitivi.Pitivi.json [hunk 18]
```diff
@@ -0,0 +1,497 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://pypi.python.org/packages/f5/f0/9da3ef24ea7eb0ccd12430a261b66eca36b924aeef06e17147f9f9d7d310/matplotlib-2.0.2.tar.gz",
+                    "sha256": "0ffbc44faa34a8b1704bc108c451ecf87988f900ef7ce757b8e2e84383121ff1"
+                }
+            ]
+        },
+        {
+            "name": "sound-theme-freedesktop",
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: It's in the runtime since a while.

File: org.pitivi.Pitivi.json [hunk 19]
```diff
@@ -0,0 +1,497 @@
...
+                {
+                    "type": "archive",
+                    "url": "http://downloads.sourceforge.net/project/opencore-amr/vo-aacenc/vo-aacenc-0.1.3.tar.gz",
+                    "sha256": "e51a7477a359f18df7c4f82d195dab4e14e7414cbd48cf79cc195fc446850f36"
+                }
+            ]
+        },
+        {
+            "name": "x265",
+            "cmake": true,
```
Comment (Barthalion) [inline org.pitivi.Pitivi.json]: Please use `"buildsystem": "cmake"`, `cmake` as key is marked as deprecated.

File: org.pitivi.Pitivi.json (hunk 9 above)
Comment (Barthalion) [reply org.pitivi.Pitivi.json]: Bump as it was rebased to the 3.24 Sdk.

File: org.pitivi.Pitivi.json (hunk 6 above)
Comment (Barthalion) [reply org.pitivi.Pitivi.json]: Bump as it was rebased to the 3.24 Sdk.

**Comment by Barthalion** [comment]
Just know when we're about to switch to 3.26! :)

We need some things to be changed, but I started a test build though: https://flathub.org/builds/#/builders/1/builds/451

File: org.pitivi.Pitivi.json (hunk 9 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: We need our own build of Gst as we have backported patches that are not in the 1.12 branch.

File: org.pitivi.Pitivi.json (hunk 11 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Done.

File: org.pitivi.Pitivi.json (hunk 13 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Done.

File: org.pitivi.Pitivi.json (hunk 12 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Well, without it Pitivi doesn't start and we get:

`Failed to register: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: org.freedesktop.DBus.Error.ServiceUnknown`

but I must be missing something, couldn't figure out what :-)

File: org.pitivi.Pitivi.json (hunk 14 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Removed for releases.

File: org.pitivi.Pitivi.json (hunk 15 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Remains from the past :-)

File: org.pitivi.Pitivi.json (hunk 16 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Removed too.

File: org.pitivi.Pitivi.json (hunk 18 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Removed.

File: org.pitivi.Pitivi.json (hunk 19 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Done.

**Comment by thiblahute** [comment]
> Just now when we're about to switch to 3.26! :)

Well, Pitivi 1.0 is not going to be based on it ;)

File: org.pitivi.Pitivi.json (hunk 17 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: He suggested the trick :-)

**Comment by TingPing** [comment]
Looks like `matplotlib` fails.

**Comment by thiblahute** [comment]
To me the issue is the usage of `build-args`, which I am removing now.

**Comment by thiblahute** [comment]
Can you trigger a rebuild please?

**Comment by TingPing** [comment]
https://flathub.org/builds/#/builders/1/builds/453

**Comment by thiblahute** [comment]
OK, works only on  64bits whch is the only one I care about for now (only tested one), I added a flathub.json so that it is the only one built for now, is that ok?

**Comment by AsavarTzeth** [comment]
You would need to fix **lame** not building on all architectures the same way as [com.github.JannikHv.Gydl](https://github.com/flathub/com.github.JannikHv.Gydl) and [org.audacityteam.Audacity](https://github.com/flathub/org.audacityteam.Audacity).

This will be the third application that has to solve the same issue. I will hence repeat my suggestion to make it a shared module.

**Comment by TingPing** [comment]
@AsavarTzeth, Seems acceptable, want to open a PR against flathub/shared-modules

**Comment by AsavarTzeth** [comment]
@TingPing Sure, I could do it right away. Should not take long.

**Comment by thiblahute** [comment]
What about building only x64-64 for now?

**Comment by TingPing** [comment]
It is acceptable for now, you probably still want to use the lame version just added to shared-modules though.

File: org.pitivi.Pitivi.json [hunk 20]
```diff
@@ -0,0 +1,462 @@
...
+        },
+        {
+            "name": "gstreamer",
+            "buildsystem": "meson",
+            "builddir": true,
+            "config-opts": ["-Ddisable_gtkdoc=true", "--libdir=lib"],
+            "sources": [
+                {
+                    "type": "git",
+                    "branch": "1.12.3",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: If you want 1.12 now the 3.26 runtime is out, so why wouldn't you use that.

File: org.pitivi.Pitivi.json [hunk 21]
```diff
@@ -0,0 +1,462 @@
...
+                    "commit": "5caeea2b0576a7e5d5463377f4c1e6d2605feab1",
+                    "url": "https://github.com/pitivi/gst-transcoder"
+                }
+            ]
+        },
+        {
+            "name": "pitivi",
+            "buildsystem": "meson",
+            "builddir": true,
+            "config-opts": ["--libdir=lib"],
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: You can drop all of these `--libdir=lib` now, this was patched in the Sdk.

File: org.pitivi.Pitivi.json [hunk 22]
```diff
@@ -0,0 +1,462 @@
...
+                "--disable-nvenc",
+                "--disable-mmal",
+                "--disable-omx",
+                "--disable-omx-rpi",
+                "--disable-cuda",
+                "--disable-cuvid",
+                "--disable-libmfx",
+                "--disable-libnpp",
+                "--disable-iconv",
+                "--disable-jni",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: It might be more manageable for you to pass `--disable-everything`.

File: org.pitivi.Pitivi.json [hunk 23]
```diff
@@ -0,0 +1,462 @@
...
+                {
+                    "type": "archive",
+                    "url": "https://pypi.python.org/packages/f5/f0/9da3ef24ea7eb0ccd12430a261b66eca36b924aeef06e17147f9f9d7d310/matplotlib-2.0.2.tar.gz",
+                    "sha256": "0ffbc44faa34a8b1704bc108c451ecf87988f900ef7ce757b8e2e84383121ff1"
+                }
+            ]
+        },
+        {
+            "name": "gsound",
+            "ensure-writable": ["easy-install.pth"],
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: I imagine this doesn't need to be here.

File: org.pitivi.Pitivi.json [hunk 24]
```diff
@@ -0,0 +1,462 @@
+{
+    "app-id": "org.pitivi.Pitivi",
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.24",
+    "command": "pitivi",
+    "finish-args": ["--socket=x11",
+                    "--socket=pulseaudio",
+                    "--socket=wayland",
+                    "--socket=session-bus",
```
Comment (TingPing) [inline org.pitivi.Pitivi.json]: No reason to pass through entire bus.

File: org.pitivi.Pitivi.json (hunk 20 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: "We need our own build of Gst as we have backported patches that are not in the 1.12 branch and we build quite many more plugins than what is provided in the Sdk."

File: org.pitivi.Pitivi.json (hunk 21 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: There is only here, without passing it explicitly we get issues, see https://phabricator.freedesktop.org/rPTVf5b106eec08f35d29d33ab75db0f5fcbe755f915

File: org.pitivi.Pitivi.json (hunk 22 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: I am just copy/pasting what is done in the `gst-libav` statically build `ffmpeg` submodule.

File: org.pitivi.Pitivi.json (hunk 23 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: Fixed.

File: org.pitivi.Pitivi.json (hunk 24 above)
Comment (thiblahute) [reply org.pitivi.Pitivi.json]: https://github.com/flathub/flathub/pull/52#discussion_r140670984

File: org.pitivi.Pitivi.json (hunk 12 above)
Comment (TingPing) [reply org.pitivi.Pitivi.json]: Probably just patch this to use the correct app-id: https://github.com/GNOME/pitivi/blob/ed321acfa2e84b67ce2ff7754ad135674b0aa74a/pitivi/application.py#L72

**Comment by thiblahute** [comment]
Ping?

**Comment by AsavarTzeth** [comment]
You made a typo:

```
Can't parse 'org.pitivi.Pitivi.json': <data>:466:18: Parse error: unexpected character `,', expected character `]'
```

**Comment by thiblahute** [comment]
Sorry about that! I am now using `lame` from the submodule and fix that issue.

**Comment by Barthalion** [comment]
https://flathub.org/builds/#/builders/1/builds/546

**Comment by TingPing** [comment]
Repository has been created: https://github.com/flathub/org.pitivi.Pitivi

**Pull request closed without merging**